### PR TITLE
cargo clippy fixes

### PR DESCRIPTION
**Pull request by haraldh**
```
$ cargo +nightly make clippy
```

### CODE REVIEW

File: mmap/src/lib.rs [hunk 1]
```diff
@@ -28,6 +34,11 @@ impl Drop for Unmap {
     }
 }
 
+/// mmap a memory region
+///
+/// # Safety
+///
+/// FIXME: yeah, why is this unsafe? The usage is unsafe, but IMHO not this operation
```
Comment (haraldh) [inline mmap/src/lib.rs:41]: @connorkuehl ?

Comment (haraldh) [reply mmap/src/lib.rs]: fixed

File: sgx/src/lib.rs [hunk 2]
```diff
@@ -78,7 +78,10 @@ pub mod crypto;
 
 #[cfg(feature = "std")]
 pub mod attestation_types;
+
 #[cfg(feature = "std")]
+// FIXME:
+#[allow(clippy::module_inception)]
```
Comment (haraldh) [inline sgx/src/lib.rs:84]: @npmccallum clippy does not like it:
https://rust-lang.github.io/rust-clippy/master/index.html#module_inception

File: sgx/src/lib.rs [hunk 3]
```diff
@@ -78,7 +78,10 @@ pub mod crypto;
...
 pub mod enclave;
```
Comment (haraldh) [inline sgx/src/lib.rs:85]: @npmccallum clippy does not like it:
https://rust-lang.github.io/rust-clippy/master/index.html#module_inception

Comment (haraldh) [reply sgx/src/lib.rs:85]: There is `src/enclave/enclave.rs`.

File: sevctl/src/main.rs [hunk 4]
```diff
@@ -22,18 +22,18 @@ const VERSION: &str = env!("CARGO_PKG_VERSION");
...
     }
 
     let mut buf = Vec::new();
     rsp.copy_to(&mut buf)
-        .expect(&format!("unable to complete {} download", usage));
+        .unwrap_or_else(|_| panic!("unable to complete {} download", usage));
 
     sev::Certificate::decode(&mut &buf[..], ())
-        .expect(&format!("unable to parse downloaded {}", usage))
+        .unwrap_or_else(|_| panic!("unable to parse downloaded {}", usage))
```
Comment (connorkuehl) [inline sevctl/src/main.rs:36]: Does clippy still complain if we don't borrow the original `format!()` statement? I feel like this is precisely the use-case for `expect()`

File: enarx-keep-sev-shim/src/payload.rs [hunk 5]
```diff
@@ -54,7 +55,7 @@ lazy_static! {
     pub static ref NEXT_MMAP_RWLOCK: RwLock<VirtAddr> = {
         let boot_info = BOOT_INFO.read().unwrap();
 
-        let mmap_start = (*PAYLOAD_VIRT_ADDR.read().deref() + (boot_info.code.end - boot_info.code.start)).align_up(4096u64);
+        let mmap_start = (*PAYLOAD_VIRT_ADDR.read().deref() + (boot_info.code.end.checked_sub(boot_info.code.start).unwrap())).align_up(4096u64);
```
Comment (connorkuehl) [inline enarx-keep-sev-shim/src/payload.rs:58]: Nit: replace `4096` with `memory::Page::size()`.

File: sevctl/src/main.rs (hunk 4 above)
Comment (haraldh) [reply sevctl/src/main.rs:36]: > Does clippy still complain if we don't borrow the original `format!()` statement? I feel like this is precisely the use-case for `expect()`

```
error[E0308]: mismatched types
  --> src/main.rs:25:30
   |
25 |     let mut rsp = rsp.expect(format!("unable to parse downloaded {}", usage));
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found struct `std::string::String`
   |
```

Comment (haraldh) [reply sevctl/src/main.rs:36]: but thinking more about it, with `unwrap_or_else` we should also output the error. Will fix.

File: enarx-keep-sev-shim/src/payload.rs (hunk 5 above)
Comment (haraldh) [reply enarx-keep-sev-shim/src/payload.rs:58]: right

**Comment by haraldh** [comment]
@connorkuehl fixed

**Comment by haraldh** [comment]
> Will you cherry pick this or just copy-paste the contents so that our CI lints everything automatically:
> 
> [connorkuehl@a9c73d7](https://github.com/connorkuehl/enarx/commit/a9c73d7be8777420877a6fcede13cc7021733d60)

oh, right forgot this... added

**Comment by haraldh** [comment]
Filed a cargo-make issue: https://github.com/sagiegurari/cargo-make/issues/447

**Pull request merged**